src/components/tabs/tabs.vue

```diff
@@ -0,0 +1,170 @@
+                </a>
+            </li>
+        </ul>
+        <div class="ui-tabs--body" ref="content">
+            <slot/>
+        </div>
+    </div>
+</template>
+
+<script>
+    import { bemMixin } from '../../mixins';
+    
+    export default {
+        components: {
+            //If we want to render not just strings we need to render The label as a Vue Component
+            RenderLabel: {
+                functional: true,
+                render: (h, ctx) => ctx.props.label,
+            },
+        },
+        mixins: [
+            bemMixin('ui-tabs'),
+        ],
+        props: {
+            selectedTab: {
+                type: Number || null,
+                default: null,
+            },
+            hasAnimation: {
+                type: Boolean,
+                default: true,
+            },
+            renderContentAlways: {
+                type: Boolean,
+                default: false,
+            },
+        },
+        data() {
+            return {
+                initTabs: [],
+                tabs: [],
+                state: {
+                    // We need the corresponding tab from tabs to set the Body active
+                    // We need the element to set to identify the active tab
+                    isEntering: {
+                        tab: null,
+                        el: null,
+                    },
+                    isLeaving: {
+                        tab: null,
+                    },
+                },
+            };
+        },
+        computed: {
+            setTabs() {
+                if (this.tabs.length > 0) {
+                    return this.tabs;
+                }
+                this.initTabs.forEach(tab => {
+                    if (tab.tabId === this.selectedTab && this.selectedTab) {
+                        tab.isActive = true;
+                    }
+                    tab.renderAlways = this.renderContentAlways;
+                    if (tab.hasAnimation === null) {
+                        tab.setHasAnimation = this.hasAnimation;
+                    }
+                });
+                
+                this.setContentHeight();
+                return this.initTabs;
+            },
+            getActiveItem() {
+                return this.tabs.find(tab => tab.isActive);
+            },
+            calcHeight() {
+                if (this.state.isEntering.tab === null) {
+                    return 0;
+                }
+                return this.state.isEntering.tab.$el.offsetHeight;
+            },
+        },
+        methods: {
+            setContentHeight() {
+                const { content } = this.$refs;
+                
+                if (!content) {
+                    return;
+                }
+                
+                this.$nextTick(() => {
+                    content.style.height = `${ this.calcHeight }px`;
+                });
+            },
+            activateTab(e, show) {
```

Comment:
I would refrain from using verbs to describe object values. Both `show` and `hide` variables point at objects. I know it's a bit of a chore, but you really should choose more verbose variable names. It immensely helps communicating your intentions. Suggestion:
* `show`: `nextVisibleTab`
* `hide`: `currentlyVisibleTab`

Reply:
changed as requested